### PR TITLE
Updating ClosureCompiler to support option banners.

### PR DESCRIPTION
**Pull request by abritinthebay**
When compressing production code it is often important (or legally required) to add in a copyright or attribution notice. This allows one to be easily set in a compatible way with many other grunt plugins - using a banner variable in the options.

This feature was very important for my companies production code, and also for personal projects - so I patched it in.

### CODE REVIEW

**Comment by dcodeIO** [comment]
Wouldn't it be sufficient to use @license or @preserve ?

``` js
/**
 * @preserve Keep this comment
 */
```

**Comment by abritinthebay** [comment]
Not really, it only allows _some_ of what this does. 

1) that would require putting syntax that may be unwanted into the source file - basically changing the code to suit the plugin, which is kind of backward process wise. 

2) Not to mention that only allows comments to be in the source - if you wanted a dynamic header that won't work (we're in grunt after all)

3) Also this option being compatibility with a lot of other plugins - like uglify, gcc, etc. means it can work almost as a drop in replacement.

Basically - it adds functionality and options with providing no downside :)

**Comment by dcodeIO** [comment]
Mhh... and what about using the `--output_wrapper "yourheader %output%"` parameter when calling the compiler?

**Comment by abritinthebay** [comment]
For a specific use case (that I required): adding a dynamic build date

**Comment by abritinthebay** [comment]
But why do that vs this implementation? No need to involve the compiler at all.

The compiler is irrelevant in this situation.

**Comment by abritinthebay** [comment]
I don't really care about my specific implementation - just that it allow an option that can be set in the grunt config options object that shows up in the output file.

That's the key feature. If you want to implement it another way it's your choice but this way works and is done already ;)

**Comment by dcodeIO** [comment]
I see, merged and thanks! :)
